### DISCUSSION TITLE
WebView MapControl instead of a port of the UWP version?

### DISCUSSION THREAD

**Discussion by eduardobragaxz** [microsoft/microsoft-ui-xaml#9323] [General]
It seems the new MapControl in 1.5 is made with WebView instead of it being a port of the one available in UWP. I just wanted to share my confusion with this decision. And I get a little concerned thinking about future controls. What's the criteria? Will there be more WebView "controls"? Why have any at all?

It's just not something I expect when using WinUI. It feels so out of place.

**Comment by dotMorten** [comment]
If the choice is between a webview control or no control due to resource constraints, perhaps getting a webview version is better than nothing?
My biggest concerns though is we lost the ability to use offline cache and the touch and mouse interactions seem glitchy and are often skipping inputs, sometimes causing giant pan operations half way around the world and sometimes gets stuck in pinch gesture. I'm not seeing happen with 3rd party native winui map controls so it's either caused by the webview or the JS API.

**Reply by eduardobragaxz** [reply]
Is that the choice, though? We know they have no problem delaying things. Ultimately, I'm concerned about setting a precedent.

**Reply by dotMorten** [reply]
> We know they have no problem delaying thing

Which is exactly why I think there's a resource constraint. I wouldn't be surprised if the UWP control was also built by their bing maps/Azure maps people and not by the uwp team

**Reply by eduardobragaxz** [reply]
If that's the case, then it would probably be a one off type thing, right? That would make me a little less apprehensive.

**Reply by bkudiess** [reply]
@dotMorten can you open an issue with details on how to repro the glitchy touch so we can keep track of it?

**Comment by mfeingol** [comment]
I for one care less about the underlying tech than about what's missing functionality-wise. Pretty much none of the functions my app uses are present in this new control. I wrote about this here: https://github.com/microsoft/microsoft-ui-xaml/issues/3024#issuecomment-1932950478

**Comment by ranjeshj** [comment]
@eduardobragaxz @dotMorten @mfeingol Thank you for the feedback. There are no near-term plans to create a native maps component. Azure maps has a rich set of features already and will continue to improve. The WinUI map control is a thin wrapper on top of that to make it easier to consume from WinUI.  This is a first step and the map control currently has a small subset of the features. This happens to be the most used set of features. We hope to continue brining more features/APIs in the future. This is a specific decision for maps and not a general pattern that we plan on doing for most controls.

**Reply by PatGet** [reply]
My main concern here: I'm paying big, multi years corporate bing maps enterprise license to use the uwp map control with xamarin and now I need (additional) azure maps keys.

**Reply by mfeingol** [reply]
@ranjeshj: thanks for following up. My main concern is the dramatic loss of functionality vs. the UWP control. MapTileSource is probably the most painful one. The more fine-grained aspects of clicking on map elements is probably a second.

In short, while most usage might be simple, some of us are using MapControl for a lot more than just displaying a few pins... I can point you to my app if you're curious.

**Reply by moshegutman** [reply]
Exactly! This just tells me that Bing Maps Enterprise license users have no future with this platform. Azure Maps is not even close to a replacement for UWP Bing MapControl

**Reply by nesherhh** [reply]
How about open sourcing UWP map control?
Bing Maps Web control is just useless for my needs.

**Reply by venklar** [reply]
@ranjeshj what's the rationale for this decision, if you don't mind me asking?

Should Windows devs wanting proper GIS features remain on WPF (which to my best knowledge has the most fully featured Maps control)?

 We've been waiting FOUR years (https://github.com/microsoft/microsoft-ui-xaml/issues/3024) for a proper control and this is what we get?! Absolutely ridiculous.

**Reply by dotMorten** [reply]
@venklar which wpf map control are you referring to?

**Reply by venklar** [reply]
@dotMorten, this one (didn't even know about the UWP wrapper they did later!):

https://learn.microsoft.com/en-us/previous-versions/bing/wpf-control/hh750210(v%3dmsdn.10)

I believe this is the control that powers MS Maps app, which you can find on the Store.

**Reply by nesherhh** [reply]
UWP is not a wrapper. And the Map App in Store is driven by this UWP control. Unfortunately it is not usable in the latest .Net

**Reply by venklar** [reply]
I meant the WPF and WinForms map controls that are wrappers of the UWP one. I think those are the currently supported ones, after end of support for Bing Maps control.

Thought the Maps app used the Bing control at some point but not sure.

Been a while since I looked at it but if memory serves me well the UWP control lacked a number of features compared with the Bing one.
